frontend/integration-tests/views/sidenav.view.ts

```diff
@@ -1,10 +1,14 @@
 import { $$, by, browser, element, ExpectedConditions as until } from 'protractor';
 
-export const navSectionFor = (name: string) => element(by.cssContainingText('.pf-c-nav > .pf-c-nav__list > .pf-c-nav__item', name));
+export const navSectionFor = (name: string) => element(by.linkText(name));
 
 export const clickNavLink = (path: [string, string]) => browser.wait(until.visibilityOf(navSectionFor(path[0])))
-  .then(() => navSectionFor(path[0]).click())
-  .then(() => browser.wait(until.visibilityOf(navSectionFor(path[0]).element(by.linkText(path[1])))))
-  .then(() => navSectionFor(path[0]).element(by.linkText(path[1])).click());
+  .then(() => navSectionFor(path[0]).getAttribute('aria-expanded').then(function(value) {
```

Comment:
Would it be cleaner to check if `path[1]` is already visible instead of checking `aria-expanded`?

Reply:
Could you give an example of using visibilityOf without browser.wait(until.visibilityOf ... 
I am not so familiar with that

Reply:
I think I found it:

.then(() => navSectionFor(path[1]).isDisplayed().then(function (isVisible) {
    if (!isVisible) {
        navSectionFor(path[0]).click();
    }
  }))